slice.go

```diff
@@ -143,3 +143,32 @@ func FilterSlice[T any](slice []T, filterFunction func(T) bool) []T {
 	}
 	return filteredSlice
 }
+
+// HasOrderedSubsetComparator compares two slices, a subset and a superset,
+// and compares them based on the compare function given, to test if
+// the subset is in order and is fully contained by the superset.
+func HasOrderedSubsetComparator[T any](superset, subset []T, compare func(T, T) bool) bool {
+	subI := 0
+	for i := 0; i < len(superset) && subI < len(subset); i++ {
```

Comment:
In this case, if we had superset [1, 2] and subset [1]. j will increment to 1 and cause a index out of bounds exception since the subset only has 1 element at the 0 index. I'll add the quick check but I the condition in the for loop needs to be there (although there may be a nicer looking replacement)